src/libraries/Microsoft.Extensions.Hosting/src/HostBuilder.cs

```diff
@@ -88,7 +88,7 @@ public IHostBuilder ConfigureServices(Action<HostBuilderContext, IServiceCollect
         /// <typeparam name="TContainerBuilder">The type of the builder to create.</typeparam>
         /// <param name="factory">A factory used for creating service providers.</param>
         /// <returns>The same instance of the <see cref="IHostBuilder"/> for chaining.</returns>
-        public IHostBuilder UseServiceProviderFactory<TContainerBuilder>(IServiceProviderFactory<TContainerBuilder> factory!!)
+        public IHostBuilder UseServiceProviderFactory<TContainerBuilder>(IServiceProviderFactory<TContainerBuilder> factory!!) where TContainerBuilder : notnull
```

Comment:
Is this a breaking change?
A public  method being more constrained than before?

Reply:
Constraint only applies to type's nullability, so it's not a breaking change (or it's on the same level as annotating some parameter as non-nullable)